packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/types/load_request_params.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+import 'package:flutter/foundation.dart';
+import 'package:webview_flutter_platform_interface/src/v4/webview_controller_delegate.dart';
+
+/// Defines the supported HTTP methods for loading a page in
+/// [WebViewControllerDelegate].
+enum LoadRequestMethod {
+  /// HTTP GET method.
+  get,
+
+  /// HTTP POST method.
+  post,
+}
+
+/// Extension methods on the [LoadRequestMethod] enum.
+extension LoadRequestMethodExtensions on LoadRequestMethod {
+  /// Converts [LoadRequestMethod] to [String] format.
+  String serialize() {
+    switch (this) {
+      case LoadRequestMethod.get:
+        return 'get';
+      case LoadRequestMethod.post:
+        return 'post';
+    }
+  }
+}
+
+/// Defines the parameters that can be used to load a page with the [WebViewControllerDelegate].
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class.
+///
+/// {@tool sample}
+/// This example demonstrates how to extend the [LoadRequestParams] to
+/// provide additional platform specific parameters.
+///
+/// When extending [LoadRequestParams] additional parameters should always
+/// accept `null` or have a default value to prevent breaking changes.
+///
+/// ```dart
+/// class AndroidLoadRequestParamsDelegate extends LoadRequestParamsDelegate {
+///   AndroidLoadRequestParamsDelegate({
+///     required Uri uri,
+///     required LoadRequestMethod method,
+///     required Map<String, String> headers,
+///     Uint8List? body,
+///     this.historyUrl,
+///   }) : super(
+///     uri: uri,
+///     method: method,
+///     body: body,
+///   );
+///
+///   final Uri? historyUrl;
+/// }
+/// ```
```

Comment:
Updated the docs

```suggestion
/// ```dart
/// class AndroidLoadRequestParams extends LoadRequestParams {
///   AndroidLoadRequestParams._({
///     required LoadRequestParams params,
///     this.historyUrl,
///   }) : super(
///     uri: params.uri,
///     method: params.method,
///     body: params.body,
///     headers: params.headers,
///   );
///
///   factory AndroidLoadRequestParams.fromLoadRequestParams(
///     LoadRequestParams params, {
///     Uri? historyUrl,
///   }) {
///     return AndroidLoadRequestParams._(params, historyUrl: historyUrl);
///   }
///
///   final Uri? historyUrl;
/// }
/// ```
```

Reply:
Fixed.